src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -2220,7 +2220,7 @@ public Lookup defineHiddenClassWithClassData(byte[] bytes, Object classData, boo
                 throw new IllegalAccessException(this + " does not have full privilege access");
             }
 
-            return makeHiddenClassDefiner(bytes.clone(), Set.of(options), false)
+            return makeHiddenClassDefiner(bytes.clone(), false, options.clone())
```

Comment:
Why do we need to clone here, but not for the method above? `makeHiddenClassDefiner` seems to always turn the options in flags right away, so I don't think we need to clone?

Reply:
I started out adding the .clone() as a standard precaution, analyzed to make sure it wasn't needed then removed it in one out of two places. Now both removed.

Reply:
The public methods used to throw `IllegalArgumentException` when duplicate class options were passed though, as a result of using [`Set.of(…)`].

[`Set.of(…)`]: https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/util/Set.html#of(E...)

Reply:
Good note, we might check the option bit is unset before bitwise-or the option bit, or remove this check (this behavior is not specified by this API and might not be relied on)

Reply:
While the public API specifies IAE to be thrown but not for the case of providing duplicate class options, it would still be behavior change that would require a CSR. And fail-fast is good. I'll add a `Set.of(options)` in the public methods to effect the invariants.